src/Workspaces/Core/Portable/Shared/Extensions/IPropertySymbolExtensions.cs

```diff
@@ -34,15 +35,18 @@ public static IPropertySymbol RenameParameters(this IPropertySymbol property, IL
         }
 
         public static IPropertySymbol RemoveAttributeFromParameters(
-            this IPropertySymbol property, INamedTypeSymbol attributeType)
+            this IPropertySymbol property, INamedTypeSymbol[] attributesToRemove)
         {
-            if (attributeType == null)
+            if (attributesToRemove == null)
             {
                 return property;
             }
 
+            Func<AttributeData, bool> shouldRemoveAttribute = a =>
+                attributesToRemove.Where(attr => attr != null).Any(attr => attr.Equals(a.AttributeClass));
+
             var someParameterHasAttribute = property.Parameters
-                .Any(p => p.GetAttributes().Any(a => a.AttributeClass.Equals(attributeType)));
```

Comment:
i feel like you don't even need to extract out the lambda.  just do: ```p.GetAttributes().Any(a => attributesToRemove.Contains(a.AttributeClass))```

But that's just me.